TestModels/Positional/src/SimplePositionalImpl.dfy

```diff
@@ -31,20 +29,36 @@ module SimplePositionalImpl refines AbstractSimplePositionalOperations {
     return Success(result);
   }
 
-
   predicate GetResourcePositionalEnsuresPublicly(input: string , output: Result<ISimpleResource, Error>)
   {true}
 
-
-
   // @positional allows use to accept the input parameters directly without the input structure
   method GetResourcePositional ( config: InternalConfig , input: string )
     returns (output: Result<ISimpleResource, Error>)
-
   {
     var resource := new SimpleResource.SimpleResource(input);
 
     // @positional allows use to return the result without the output structure
     return Success(resource);
   }
+
+  predicate GetIntegerInputPositionEnsuresPublicly(input: int32 , output: Result<GetIntegerInputPositionOutput, Error>)
+  {true}
+
+  method GetIntegerInputPosition ( config: InternalConfig , input: int32 )
+    returns (output: Result<GetIntegerInputPositionOutput, Error>)
+  {
+    var res := GetIntegerInputPositionOutput(value := Some(input));
+    return Success(res);
+  }
+
+  predicate GetIntegerOutputPositionEnsuresPublicly(input: GetIntegerOutputPositionInput , output: Result<int32, Error>)
+  {true}
+
+  method GetIntegerOutputPosition ( config: InternalConfig , input: GetIntegerOutputPositionInput )
```

Comment:
I think we can do better naming. It's confusing to read `input: GetIntegerOutputPositionInput` , maybe something like `GetIntegerPositionalAsResponse`